pkg/trace/agent/agent.go

```diff
@@ -193,6 +208,9 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 
 		// Extra sanitization steps of the trace.
 		for _, span := range t {
+			for k, v := range a.conf.AdditionalMetadata {
+				traceutil.SetMeta(span, k, v)
```

Comment:
Do you want this on every span of the trace? Why not only the root? It's totally up to you but just making sure this is what you want :)

Reply:
Yeah, we need to apply __dd.origin:lambda to every span in the trace. It enables special processing/displaying on the trace intake side.